File: example/ios/CounterViewManager.m
//
//  CounterViewManager.m
//  RnKalturasExample
//
//  Created by Sanjay Achar on 18/04/21.
//

#import "React/RCTViewManager.h"
@interface RCT_EXTERN_MODULE(CounterViewManager, RCTViewManager)

// Initial Props
RCT_EXPORT_VIEW_PROPERTY(ServerUrl, NSString)
RCT_EXPORT_VIEW_PROPERTY(PartnerId, NSString)
RCT_EXPORT_VIEW_PROPERTY(UiConfId, NSString)

RCT_EXTERN_METHOD(
  play:(nonnull NSNumber *)node
)

RCT_EXTERN_METHOD(
  pause:(nonnull NSNumber *)node
)

RCT_EXTERN_METHOD(
  toggleFullScreen:(nonnull NSNumber *)node
)

@end

File: example/ios/CounterViewManager.swift
//
//  CounterViewManager.swift
//  RnKalturasExample
//
//  Created by Sanjay Achar on 18/04/21.
//

import Foundation

@objc(CounterViewManager)
class CounterViewManager: RCTViewManager {
  
  override func view() -> UIView! {
    return CounterView()
  }
  
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc func pause(_ node: NSNumber) {
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view(
        forReactTag: node
      ) as! CounterView
      
      component.pause()
    }
  }
  
  @objc func play(_ node: NSNumber) {
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view(
        forReactTag: node
      ) as! CounterView
      
      component.play()
    }
  }
  
  @objc func toggleFullScreen(_ node: NSNumber) {
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view(
        forReactTag: node
      ) as! CounterView
      
      component.toggleFullScreen()
    }
  }
}
